### PR TITLE
Contract sanity checking

### LINKED ISSUE #228: Add contract sanity checking during build / release
Let's add sanity checking (check exports, imports, deterministic middleware, ...) to wasm-compiled (.wasm) smart contracts.

An easy way to do this would be to add a simple / dummy integration test for the contract. Given that compiling and running integration tests is expensive, we could run these tests in the CI only during the contract building / release step.

An alternative or complement to this would be to use a small utility to just compile the contract. Something similar to [`module_size`](https://github.com/CosmWasm/cosmwasm/blob/main/packages/vm/examples/module_size.rs). Perhaps with some additional checks / validations.

Let's do this for cosmwasm-plus contracts, and also cosmwasm and cosmwasm-examples contracts.

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #228.

### CODE REVIEW

File: .circleci/config.yml
Comment (ethanfrey) [inline .circleci/config.yml:795]: Nice line.

And I like where this is placed. Doesn't add much more delay to the CI and ensures compatibility

File: .circleci/config.yml
Comment (ethanfrey) [inline .circleci/config.yml:343]: Should we remove the integration/test.rs file as well?
I would suggest that (and remove devDeps to keep CI fast)

**Review by ethanfrey — APPROVED** [review]
Looks good to merge once cosmwasm pr is merged and CI is happy

File: .circleci/config.yml
Comment (maurolacy) [reply .circleci/config.yml:795]: Yes, this is optimal in that sense. Also compiling this utility for debug, which is twice as fast.

File: .circleci/config.yml
Comment (maurolacy) [reply .circleci/config.yml:343]: I already removed the integ tests, but forgot about the dev deps. Thanks.

**Pull request merged**
